src/openvic-simulation/research/Invention.cpp

```diff
@@ -0,0 +1,63 @@
+#include "Invention.hpp"
+#include <algorithm>
+#include <cstring>
+#include <utility>
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+Invention::Invention(std::string_view identifier, ModifierValue&& values, bool news, unit_set_t activated_units, building_set_t activated_buildings, bool unlock_gas_attack, bool unlock_gas_defence) : Modifier { identifier, std::move(values), 0 }, news { news }, activated_units { std::move(activated_units) }, activated_buildings { std::move(activated_buildings) }, unlock_gas_attack { unlock_gas_attack }, unlock_gas_defence { unlock_gas_defence } {} //TODO icon
+
+InventionManager::InventionManager() : inventions { "inventions" } {}
+
+bool InventionManager::add_invention(std::string_view identifier, ModifierValue&& values, bool news, Invention::unit_set_t activated_units, Invention::building_set_t activated_buildings, bool unlock_gas_attack, bool unlock_gas_defence) {
+	if (identifier.empty()) {
+		Logger::error("Invalid invention identifier - empty!");
+		return false;
+	}
+
+	return inventions.add_item({ identifier, std::move(values), news, std::move(activated_units), std::move(activated_buildings), unlock_gas_attack, unlock_gas_defence });
+}
+
+bool InventionManager::load_inventions_file(ModifierManager const& modifier_manager, UnitManager const& unit_manager, BuildingManager const& building_manager, ast::NodeCPtr root) {
+	return expect_dictionary_reserve_length(inventions, [this, &modifier_manager, &unit_manager, &building_manager](std::string_view identifier, ast::NodeCPtr value) -> bool {
+		ModifierValue loose_modifiers;
+		ModifierValue modifiers;
+
+		Invention::unit_set_t activated_units;
+		Invention::building_set_t activated_buildings;
+		Invention::crime_set_t enabled_crimes;
+
+		bool unlock_gas_attack = true;
+		bool unlock_gas_defence = true;
```

Comment:
```suggestion
		bool unlock_gas_attack = false;
		bool unlock_gas_defence = false;
```
Shouldn't these default to `false`?